tests/test_enterprise/api/test_views.py

```diff
@@ -4436,6 +4427,84 @@ def test_bulk_enrollment_in_bulk_courses_nonexisting_user_id(
         # no notifications to be sent unless 'notify' specifically asked for in payload
         mock_notify_task.assert_not_called()
 
+    @ddt.unpack
+    @mock.patch("enterprise.api.v1.views.enrollment_api")
+    @mock.patch('enterprise.api.v1.views.get_best_mode_from_course_key')
+    @mock.patch("enterprise.utils.lms_enroll_user_in_course")
+    def test_bulk_enrollment_enroll_after_cancel(
+        self,
+        mock_platform_enrollment,
+        mock_get_course_mode,
+        mock_enrollment_api,
+    ):
+        """
+        Test that even after a cancelled enterprise enrollment, an attempt to re-enroll the same learner in content
+        results in expected state and payload.
```

Comment:
High level architecture of this test: I use factories to create the enrollment initially, then use 2 APIs to subsequently cancel then re-enroll.